monix-tail/shared/src/main/scala/monix/tail/internal/IterantInterleave.scala

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2014-2018 by The Monix Project Developers.
+ * See the project homepage at: https://monix.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package monix.tail.internal
+
+import cats.effect.Sync
+import monix.tail.Iterant
+
+
+private[tail] object IterantInterleave {
+  def apply[F[_], A, B >: A](lh: Iterant[F, A], rh: Iterant[F, B]) (implicit F: Sync[F]): Iterant[F, B] = {
```

Comment:
Just a small comment — you don't need the subtyping relationship to avoid covariance issues here, i.e. you don't need a `B` type parameter.

Reply:
Or in other words, it's OK to have it on the method in `Iterant`, but not on this function here.

Reply:
@alexandru If I remove `B` in `apply`:
```
def apply[F[_], A](lh: Iterant[F, A], rh: Iterant[F, A]) (implicit F: Sync[F]): Iterant[F, A]
```
I get error:
```
[error] monix/monix-tail/shared/src/main/scala/monix/tail/Iterant.scala:1528:23: type mismatch;
[error]  found   : monix.tail.Iterant[F,A]
[error]  required: monix.tail.Iterant[F,B]
[error] Note: A <: B, but class Iterant is invariant in type A.
[error] You may wish to define A as +A instead. (SLS 4.5)
[error]     IterantInterleave(self, rhs)(F)
[error]                       ^
```
Do you want to remove `B` from `Iterant.interleave`?

Reply:
Ah, true, `Iterant` is invariant for now as a design decision having to do with covariance not mixing well with higher kinds, but you can use `self.upcast[B]` in that call.

Reply:
@alexandru Thanks! Done. I saw `upcast` but did not think of using it here.